### DISCUSSION TITLE
[Share] Add `/unread` tag to new items (support multi-device syncing)

### DISCUSSION THREAD

**Discussion by windingwind** [windingwind/zotero-actions-tags#202] [Action Scripts]
### Description

Add `/unread` tag to new items (support multi-device syncing)

### Event

Creat Item

### Operation

Script

### Data

```javascript
/**
 * Add `/unread` tag to new items (support multi-device syncing)
 * @author windingwind
 * @usage See below
 * @link https://github.com/windingwind/zotero-actions-tags/discussions/202
 * @see https://github.com/windingwind/zotero-actions-tags/discussions/202
 */
if (!item) return;

if (!item.getTags().find(t => t.tag === "/read")) {
	item.addTag("/unread", 1);
	return `Add tag /unread to item ${item.getField("title")}`;
}
```

### Anything else

To support multi-device syncing, please combine this action with the following two actions and disable the other actions that add `/unread`:

Remove `/unread` after the reader is closed:

<img width="351" alt="image" src="https://github.com/windingwind/zotero-actions-tags/assets/33902321/e2463a93-7960-4545-8fc3-14895e42f8de">

Add `/read` after the reader is closed:

<img width="351" alt="image" src="https://github.com/windingwind/zotero-actions-tags/assets/33902321/25b27748-7632-4570-87a7-dc6b5a48edbd">

**Comment by fredericky123** [comment]
谢谢大佬！
我按如图都设置好了，可是新导入的item先自动添加了unread，然后怎么又自动消失了呢

**Reply by windingwind** [reply]
“自动消失”指的是什么？请详细描述从添加了标签到消失中间操作了什么，什么时候消失的

**Reply by fredericky123** [reply]
就是导入后，运行了几个插件，等看到导入的一刹那是有unread标签的，等这个item被添加完毕，格式调整正确，unread标签又被删了
暂停了Linter, Notero等,删除了图中图示代码，也依然无效，会先自动添加再自动删除
<img width="578" alt="image" src="https://github.com/windingwind/zotero-actions-tags/assets/13982749/5317daa0-560a-4fc2-b95a-34d11f93ad8a">

**Reply by windingwind** [reply]
已更新动作脚本，请覆盖重新保存该动作

**Reply by fredericky123** [reply]
感谢大佬，一切正常了！

**Comment by DonnaCoxBaker** [comment]
Today, I set up a new computer I had bought. I added Zotero and several plugins including this one, which had the default script to add the "unread" tag. When I logged into my Zotero account to bring my data in, it added /unread tags for hours--to more than 8000 items. Then it took hours for Zotero to sync all of these items, since every one of them had been altered. I'll now need to delete the tag to take it off of everything, and it will all have to sync every item again. Can something be done to prevent this happening? It seems that it should not add the tag when something comes in by syncing, since it would have already been added on the other computer?

**Reply by windingwind** [reply]
Thanks for letting me know and sorry for that. I will do a fix soon!

**Reply by DonnaCoxBaker** [reply]
Thanks so much!
